### PR TITLE
Implemented CTRL + Arrows to move to next/previous word

### LINKED ISSUE #82: Ctrl + Arrow should move to previous/next word

### PR DESCRIPTION
**Pull request by carol-mb**
Using CTRL+ Arrow Left/Right moves the cursor to previous/next word.

- I extended the fuctionality of cursor movement to detect if CTRL key is pressed.
- I updated the README.md to reflect these changes.

This closes #82

### CODE REVIEW

File: src/editor.rs [hunk 1]
```diff
@@ -187,26 +187,36 @@ impl Editor {
...
-            (AKey::Left, _) if self.cursor.y > 0 => {
-                // ← at the beginning of the line: move to the end of the previous line. The x
-                // position will be adjusted after this `match` to accommodate the current row
-                // length, so we can just set here to the maximum possible value here.
-                self.cursor.y -= 1;
-                self.cursor.x = usize::MAX;
+            (AKey::Left, Some(row)) if self.cursor.x > 0 => {
+                cursor_x -= row.get_char_size(row.cx2rx[cursor_x] - 1);
+                // ← moving to previous word
+                while ctrl && cursor_x > 0 && row.chars[cursor_x - 1] != 0x20 {
```
Comment (ilai-deutel) [inline src/editor.rs:196]: I think `b' '` would be clearer than `0x20`

**Review by ilai-deutel — CHANGES_REQUESTED** [review]
Thank you for your contribution!

**Comment by ilai-deutel** [comment]
@all-contributors please add @carol-mb for code

File: src/editor.rs [hunk 2]
```diff
@@ -910,8 +918,8 @@ mod tests {
         }
         editor.delete_char();
         assert_eq!(editor.rows[0].chars, "Hello".as_bytes());
-        editor.move_cursor(&AKey::Left);
-        editor.move_cursor(&AKey::Left);
+        editor.move_cursor(&AKey::Left, false);
+        editor.move_cursor(&AKey::Left, false);
         editor.delete_char();
         assert_eq!(editor.rows[0].chars, "Helo".as_bytes());
     }
```
Comment (ilai-deutel) [inline src/editor.rs:925]: Could you add a test for the new behavior?

File: src/editor.rs (hunk 1 above)
Comment (carol-mb) [reply src/editor.rs]: I changed `0x20` to `b' '`. Thank you for feedback!

File: src/editor.rs (hunk 2 above)
Comment (carol-mb) [reply src/editor.rs:925]: Sure! I added it.

**Comment by ilai-deutel** [comment]
Thanks again!

**Pull request merged**